### PR TITLE
Enable e2e tests to check for successful events other than `DONE`

### PR DESCRIPTION
**Pull request by Adamantios**
## Proposed changes

We now define a happy path for our e2e tests, using a `RoundChecks` dataclass. This enables us to specify the name of the successful event which previously could only be `DONE` for the test to succeed. This will also allow us to re-enable the APY e2e tests.

## Fixes

n/a

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [x] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

## Further comments

n/a

### CODE REVIEW

File: tests/test_agents/base.py [hunk 1]
```diff
@@ -129,10 +145,6 @@ def __set_configs(self, i: int, nb_agents: int) -> None:
             "vendor.valory.connections.abci.config.host",
             ANY_ADDRESS,
         )
-        self.set_config(
-            "vendor.valory.connections.abci.config.host",
-            ANY_ADDRESS,
-        )
```
Comment (Adamantios) [inline tests/test_agents/base.py:135]: Not related to this PR, I removed it because it was a duplicate.

File: tests/test_agents/base.py [hunk 2]
```diff
@@ -109,9 +124,10 @@ def set_config(
         dotted_path: str,
         value: Any,
         type_: Optional[str] = None,
+        aev: bool = True,
     ) -> Result:
         """Set config value."""
-        return super().set_config(dotted_path, value, type_, aev=True)
+        return super().set_config(dotted_path, value, type_, aev)
```
Comment (Adamantios) [inline tests/test_agents/base.py:130]: Not related to this PR, added `aev` as an argument.

File: tests/test_agents/base.py [hunk 3]
```diff
@@ -37,6 +38,21 @@
...
+    """
+    Class for the necessary checks of a round during the tests.
+
+    name: is the name of the round for which the checks should be performed.
+    event: is the name of the event that is considered as successful.
+    n_periods: is the number of periods this event should appear for the check to be considered successful.
+    """
+
+    name: str
+    success_event: str = "DONE"
```
Comment (Adamantios) [inline tests/test_agents/base.py:52]: The default success event is still `DONE`; we do not need to type it every time, except if it is different.

File: tests/test_agents/test_apy_estimation_abci.py [hunk 4]
```diff
@@ -23,27 +23,26 @@
...
+HAPPY_PATH = (
+    RoundChecks("collect_history"),
+    RoundChecks("transform"),
+    RoundChecks("preprocess"),
+    RoundChecks("randomness"),
+    RoundChecks("optimize"),
+    RoundChecks("train"),
+    RoundChecks("train", success_event="FULLY_TRAINED"),
+    RoundChecks("test"),
+    RoundChecks("estimate", n_periods=2, success_event="ESTIMATION_CYCLE"),
```
Comment (Adamantios) [inline tests/test_agents/test_apy_estimation_abci.py:40]: Here is an example of how we can use other successful events for the happy path's definition.

Comment (Karrenbelt) [reply tests/test_agents/test_apy_estimation_abci.py:40]: unrelated to this PR an sich, this sentence confuses me. What do you mean by "train on test data" after optimization?

Comment (Adamantios) [reply tests/test_agents/test_apy_estimation_abci.py:40]: Yes, I did not phrase that well. I meant simply on all the available data.

In the beginning, after optimizing, we split the data, train on the train split, and test on the test split. Then, if the performance is acceptable on the test split, we train on all the available data in order to create the final model. 

Hence, you can see why I incorrectly phrased it before.

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**